crates/shared/src/zeroex_api.rs

```diff
@@ -363,6 +365,7 @@ pub struct DefaultZeroExApi {
     client: Client,
     base_url: Url,
     block_stream: CurrentBlockStream,
+    orders_cache: Mutex<TimedCache<OrdersQuery, Vec<OrderRecord>>>,
```

Comment:
`Mutex` is used here instead of `RwLock` because `cached::TimedCache` requires a mutable reference to the cache, whereas `RwLock::read` cannot provide that. It's worth searching for another cache implementation to enable separating read/write operations and not block all the threads on reads.